tools/test.py

```diff
@@ -142,11 +169,14 @@ def main():
         if args.out:
             print(f'\nwriting results to {args.out}')
             mmcv.dump(outputs, args.out)
-        kwargs = {} if args.options is None else args.options
+        kwargs = {} if args.eval_options is None else args.eval_options
         if args.format_only:
             dataset.format_results(outputs, **kwargs)
         if args.eval:
-            dataset.evaluate(outputs, args.eval, **kwargs)
+            eval_kwargs = cfg.get('evaluation', {}).copy()
+            eval_kwargs.pop('interval', None)
+            eval_kwargs.update(dict(metric=args.eval, **kwargs))
```

Comment:
It's a bit hard-coded, and it may also trigger some bugs if there are some other arguments specified in `evaluation=dict()` in the config.

Reply:
In this case, I suggest adding poping `interval`, `tmpdir`, `start` and `gpu_collect`   keys. `start` is an additional key to be added in EvalHook (https://github.com/open-mmlab/mmdetection/pull/3550)